samples/net/rpl-node/src/main.c

```diff
@@ -0,0 +1,1164 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if 1
+#define SYS_LOG_DOMAIN "rpl-node"
+#define NET_SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#define NET_LOG_ENABLED 1
+#endif
+
+#include <zephyr.h>
+#include <errno.h>
+#include <board.h>
+
+#include <misc/byteorder.h>
+#include <net/net_core.h>
+#include <net/net_ip.h>
+#include <net/net_pkt.h>
+#include <net/net_context.h>
+
+#include <net_private.h>
+#include <rpl.h>
+#include <icmpv6.h>
+#include <6lo.h>
+#include <ipv6.h>
+
+#include <gpio.h>
+
+#include <net/coap.h>
+#include <net/coap_link_format.h>
+
+#if defined(CONFIG_NET_L2_IEEE802154)
+#include <ieee802154_settings.h>
+#endif
+
+#define MY_COAP_PORT 5683
+
+#define ALL_NODES_LOCAL_COAP_MCAST					\
+	{ { { 0xff, 0x02, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0xfd } } }
+
+#if defined(LED0_GPIO_PORT)
+#define LED_GPIO_NAME LED0_GPIO_PORT
+#define LED_PIN LED0_GPIO_PIN
+#else
+#define LED_GPIO_NAME "(fail)"
+#define LED_PIN 0
+#endif
+
+#define RPL_MAX_REPLY 75
+
+#define PKT_WAIT_TIME K_SECONDS(1)
+
+static struct net_context *context;
+static struct device *led0;
+static const u8_t plain_text_format;
+
+/* LED */
+static const char led_on[] = "LED0 ON";
+static const char led_off[] = "LED0 OFF";
+static const char led_toggle_on[] = "LED Toggle ON";
+static const char led_toggle_off[] = "LED Toggle OFF";
+
+static bool fake_led;
+
+/* RPL */
+static const char rpl_parent[] = "RPL Parent:";
+static const char rpl_no_parent[] = "No parent yet";
+
+static const char rpl_rank[] = "RPL Rank:";
+static const char rpl_no_rank[] = "No rank yet";
+
+static const char rpl_link[] = "Link Metric:";
+static const char rpl_no_link[] = "No link metric yet";
+
+static const char ipv6_no_nbr[] = "No IPv6 Neighbors";
+
+static void get_from_ip_addr(struct coap_packet *cpkt,
+			     struct sockaddr_in6 *from)
+{
+	struct net_udp_hdr *udp;
+
+	net_ipaddr_copy(&from->sin6_addr, &NET_IPV6_HDR(cpkt->pkt)->src);
+
+	udp = (struct net_udp_hdr *)((u8_t *)(NET_IPV6_HDR(cpkt->pkt)) +
```

Comment:
Same for usage of net_udp_get_hdr()

Reply:
Sure, I will update all net_udp_get_hdr() related calls.